maker/src/connection.rs

```diff
@@ -39,13 +39,18 @@ use xtras::SendInterval;
 
 const TCP_TIMEOUT: Duration = Duration::from_secs(10);
 
-fn parse_vergen_version(vergen_string: &str) -> semver::Version {
-    let vergen = semver::Version::parse(vergen_string).expect("daemon version to parse");
+fn parse_vergen_version(vergen_string: &str) -> Result<semver::Version> {
+    let vergen = semver::Version::parse(vergen_string)
+        .context("Could not parse daemon_version: {vergen_string}")?;
```

Comment:
🙃 

It's weird that the context talks about `daemon` here, when this is just a helper that maps string to `semver::Version`.

Reply:
It's very nitpicky (hence the 🙃), but it would make more sense to me to:

- Keep the helper as a general helper called `parse_vergen_version`.
- Remove the `context` and make the helper return a `Result<semver::Version, semver::Error>`.
- Add the `context` to line 503, where the helper is called (or alternatively to the trace on line 510).

Reply:
I think what you did is fine too! Just a tiny style thing almost.